service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/HealthcareServiceMapper.java

```diff
@@ -60,15 +66,21 @@ private HealthcareService mapSingleHealthcareService(
         }
 
         if (intendedRecipient.isSetReceivedOrganization()) {
-            POCDMT000002UK01Organization receivedOrganization =
-                intendedRecipient.getReceivedOrganization();
+            POCDMT000002UK01Organization receivedOrganization = intendedRecipient.getReceivedOrganization();
             Organization organization = organizationMapper.mapOrganization(receivedOrganization);
-            healthcareService.setProvidedBy(new Reference(organization));
-            healthcareService.setProvidedByTarget(organization);
-            if (receivedOrganization.sizeOfNameArray() > 0) {
-                ON name = receivedOrganization.getNameArray(0);
-                healthcareService.setName(nodeUtil.getAllText(name.getDomNode()));
+            Coding code = new Coding().setCode(String.valueOf(informationRecipient.getTypeCode()));
+            if (code.getCode() == PRCP) {
```

Comment:
I would check if typeCode == PRCP in line 41 in `mapHealthcareService` method, before trying to map HealthcareService. From what I understand, if InformationRecipient has different typeCode, then we don't want to map it at all.